### ISSUE TITLE
FutureGrant on Database Role doesn't fully qualify the role

### ISSUE THREAD

**Issue by dnabb** [Titan-Systems/titan#162]
Hello, and thanks for the tool.

The following code:

```python
db = Database(name='whatever')
role = DatabaseRole(name='whatever_role', database=db)
fg = FutureGrant(priv='SELECT', on_type="table", in_type=db.resource_type, in_name=db.name, to=role)

bp = Blueprint(
    resources=[db, role, fg],
    dry_run=True
    )
```

Generates this plan, which appears correct:

```text
» titan core
» Plan: 3 to create, 0 to update, 0 to transfer, 0 to drop.

+ urn::XX12345:database/WHATEVER {
  + name                            = "WHATEVER"
  + transient                       = False
  + owner                           = "SYSADMIN"
  + data_retention_time_in_days     = 1
  + max_data_extension_time_in_days = 14
  + external_volume                 = None
  + catalog                         = None
  + default_ddl_collation           = None
  + comment                         = None
}

+ urn::XX12345:database_role/WHATEVER.WHATEVER_ROLE {
  + name     = "WHATEVER_ROLE"
  + database = "WHATEVER"
  + owner    = "USERADMIN"
  + comment  = None
}

+ urn::XX12345:future_grant/WHATEVER_ROLE?priv=SELECT&on=database/WHATEVER.<TABLE> {
  + priv         = "SELECT"
  + on_type      = "TABLE"
  + in_type      = "DATABASE"
  + in_name      = "WHATEVER"
  + to           = "WHATEVER.WHATEVER_ROLE"
  + grant_option = False
}
```

However, it then produces the following SQL:

```sql
USE SECONDARY ROLES ALL
USE ROLE SYSADMIN
CREATE DATABASE WHATEVER DATA_RETENTION_TIME_IN_DAYS = 1 MAX_DATA_EXTENSION_TIME_IN_DAYS = 14
USE ROLE SYSADMIN
CREATE DATABASE ROLE WHATEVER.WHATEVER_ROLE
GRANT OWNERSHIP ON DATABASE ROLE WHATEVER.WHATEVER_ROLE TO ROLE USERADMIN COPY CURRENT GRANTS
USE ROLE SECURITYADMIN'
GRANT SELECT ON FUTURE TABLES IN DATABASE WHATEVER TO ROLE WHATEVER_ROLE <<< This will fail
```

The database role doesn't get fully qualified, which causes the SQL to fail (Role 'WHATEVER_ROLE' does not exist or not authorized even though we just created it).

**Comment by dnabb** [comment]
My guess:
https://github.com/Titan-Systems/titan/blob/296b108dd9886aaee3feda539755690e45032d30/titan/lifecycle.py#L102
Should be `data["to"]`

**Comment by teej** [comment]
I can repro, I'll fix it shortly.

**Comment by teej** [comment]
Fixed in [v0.10.20](https://github.com/Titan-Systems/titan/tree/refs/tags/v0.10.20).

Also, for future grants, I recommend the more sql-like syntax:

```Python
fg = FutureGrant(priv='SELECT', on_future_tables_in=db, to=role)
```

**Comment by dnabb** [comment]
Hi,

I am not sure if this was intended, but [v0.10.20](https://github.com/Titan-Systems/titan/tree/refs/tags/v0.10.20) now actually breaks the on_type / in_type / in_name syntax:

`fg = FutureGrant(priv='SELECT', on_type="table", in_type=db.resource_type, in_name=db.name, to=role)`
returns:
`ValueError: You must specify one 'on_future_' parameter`

Is that syntax officially deprecated now or just a bug/regression?

**Comment by dnabb** [comment]
Some background on why I am still trying to use the on_type / in_type / in_name syntax:
 
I am trying to create a sort of "role template", where I define grants by object type in a "config dictionary" that might look like this:

```yaml
x_role:
  "table":
    - SELECT
    - REFERENCES
  "iceberg table":
    - SELECT
    - REFERENCES
  "dynamic table":
    - SELECT
    - MONITOR
```
.. and for each privilege & object type, I want to apply both a `FutureGrant` and a `GrantOnAll`.

I find this makes it easier to see at a glance "what is x_role supposed to do", but is a bit of an orthogonal approach to what the SQL syntax would lead to (which is one of the reasons why I am looking at a Python tool to manage it).

I can still achieve the same result by "preparing" the `f"on_{future\all}_{object_type_pluralized"}` in a dict and then pass them as **kwargs, but it looks like that is basically what `FutureGrant` and `GrantOnAll` do under the hood?

We can move this to a separate discussion (I can see how this is basically trying to recreate your example in the [modules idea](https://github.com/Titan-Systems/titan/discussions/120) with a different YAML spec), but my point is:
- The on_type / in_type / in_name syntax has its uses (at least in Python)
- If possible, it should be a first class citizen

**Comment by teej** [comment]
Sorry for this regression! It wasn't my intent to break this syntax. I'll fix today.

**Comment by carlioth** [comment]
Hey
Is there any update on this? 
I am currently awaiting this fix too.

**Comment by teej** [comment]
Fixed in [v0.11.0](https://github.com/Titan-Systems/titan/releases/tag/v0.11.0)
